File: example/spring-boot-graphql/src/main/kotlin/cz/encircled/joiner/springbootgraphql/JoinerConfig.java
package cz.encircled.joiner.springbootgraphql;

import cz.encircled.joiner.core.Joiner;
import cz.encircled.joiner.query.join.DefaultJoinGraphRegistry;
import cz.encircled.joiner.query.join.J;
import cz.encircled.joiner.springbootgraphql.model.Author;
import jakarta.persistence.EntityManager;
import jakarta.persistence.PersistenceContext;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.util.List;

import static cz.encircled.joiner.springbootgraphql.model.QPost.post;
import static cz.encircled.joiner.springbootgraphql.model.QStatus.status;

@Configuration
public class JoinerConfig {

    @PersistenceContext
    EntityManager entityManager;

    @Bean
    Joiner joiner() {
        Joiner joiner = new Joiner(entityManager);

        DefaultJoinGraphRegistry joinRegistry = new DefaultJoinGraphRegistry();
        joiner.setJoinGraphRegistry(joinRegistry);
        joinRegistry.registerJoinGraph("posts", List.of(J.left(post)), Author.class);
        joinRegistry.registerJoinGraph("posts/status", List.of(J.left(post).nested(status)), Author.class);
        joinRegistry.registerJoinGraph("status", List.of(J.inner(status)), Author.class);

        return joiner;
    }

}

File: example/spring-boot-example/src/main/kotlin/cz/encircled/joiner/springbootexample/JoinerConfig.kt
package cz.encircled.joiner.springbootexample

import cz.encircled.joiner.core.Joiner
import cz.encircled.joiner.kotlin.JoinerKt
import jakarta.persistence.EntityManager
import jakarta.persistence.PersistenceContext
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class JoinerConfig {

    @PersistenceContext
    lateinit var em: EntityManager

    @Bean
    fun joiner(): Joiner {
        return Joiner(em)
    }

    @Bean
    fun joinerKt(): JoinerKt {
        return JoinerKt(em)
    }

}